internal/controller/util/vm_util.go

```diff
@@ -57,35 +66,263 @@ func ListVMsByVMNamespace(
 	log logr.Logger,
 	vmNamespaceList []string,
 	vmList []string,
-) ([]string, error) {
-	var foundVMList []string
+) []virtv1.VirtualMachine {
+	foundVM := &virtv1.VirtualMachine{}
 
-	var notFoundErr error
+	foundVMList := make([]virtv1.VirtualMachine, 0, len(vmList))
+	for _, ns := range vmNamespaceList {
+		for _, vm := range vmList {
+			vmLookUp := types.NamespacedName{Namespace: ns, Name: vm}
+			if err := apiReader.Get(ctx, vmLookUp, foundVM); err != nil {
+				continue
+			}
+
+			foundVMList = append(foundVMList, *foundVM)
+		}
+	}
+
+	if len(foundVMList) > 0 {
+		return foundVMList
+	}
+
+	return nil
+}
+
+// IsVMDeletionInProgress returns true if any listed KubeVirt VM within the given protected NS is in deletion state.
+// Skips VMs that cannot be fetched (likely already deleted); checks all (namespace, name) pairs.
+func IsVMDeletionInProgress(ctx context.Context,
+	k8sclient client.Client,
+	vmList []string,
+	vmNamespaceList []string,
+	log logr.Logger,
+) bool {
+	log.Info("Checking if VirtualMachines are being deleted",
```

Comment:
I would potentially remove this log line, in an effort to keep logging to minimum, as this also prints the list of VMs.